### DISCUSSION TITLE
merging Pathways and Version Control

### DISCUSSION THREAD

**Discussion by alexmadsen1** [wikipathways/wikipathways-help#95] [General]
Hi @khanspers, I'm reaching out to inquire about any recent developments regarding the integration of git version control for merging pathway branches. We've seen great collaborative efforts on WP5420, which is wonderful. However, this collaboration has unintentionally overshadowed several hours of my recent work, which includes a significant cleanup of the pathway to manage its growth. Could you advise on how to merge these changes into the main branch effectively?

**Comment by khanspers** [comment]
Hi @alexmadsen1, looking at the history of WP5420 at [WP classic](https://classic.wikipathways.org/index.php/Pathway:WP5420), there are many edits by you starting in late November, I assume these correspond to your recent work you are referring to? The most recent edit, by Eweitz, appears to have been correctly made on the 127879 revision (resulting from your work on Jan 5), but perhaps I'm missing something. Can you explain? 
If the latest edit is the problem, the solution is to revert it to restore the 127879 revision, and then redo the edit to the description that was done by Eweitz (i.e. spelling out the acronym for ADHD).

To answer your question about the current status of our new system, it is still in development. With the new system, minor edits may be automatically merged, but larger edits, especially edits that result in a conflict, will still require manual review and potentially additional revisions before it can be merged.

**Comment by alexmadsen1** [comment]
I believe the problem/split occurred with the check-in of  127881. I now
have a stranded copy of 127879 plus additional updates on my desktop. When
I checked it in, I received a "you do not have the latest copy error. "
127881
Approved
view <https://classic.wikipathways.org/index.php/Pathway:WP5420> 02:04, 6
January 2024 Eweitz
<https://classic.wikipathways.org/index.php/User:Eweitz> Modified
description
127879 view
<https://classic.wikipathways.org/index.php?title=Pathway:WP5420&oldid=127879>
23:55,
5 January 2024 Alexmadsen1
<https://classic.wikipathways.org/index.php/User:Alexmadsen1> uodate TSC1/2
<-->AMPK <-> NOS

**Comment by khanspers** [comment]
OK, I see. As a user, your only solution would be to remake the changes on the latest revision, unfortunately. However, if you send me your latest gpml file (kristina.hanspers@gladstone.ucsf.edu), I can use it to directly update the model. This isn't always an option, for example if the latest live edits are extensive, but in this case the latest edit is limited to the description. The only drawback with this method is that the edit to incorporate your changes will not be accurately attributed to you; the best I can do is to list your username in the comment of the edit.
